### PR TITLE
Update num worker calculation due to PTL flag changes

### PR DESCRIPTION
**Pull request by redoctopus**
Signed-off-by: Jocelyn Huang <jocelynh@nvidia.com>

# What does this PR do ?

Removes conditionals in favor of simple calculation of num workers due to changes in PTL flags, as per conversation with @nithinraok and prior discussion in #3589 

# Changelog
- Update calculation of num workers in model class (optimizer setup)
- Update calculation of num workers in TTS helpers
- Bump PTL version requirement from 1.6.0 -> 1.6.1

**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/collections/tts/helpers/helpers.py [hunk 1]
```diff
@@ -91,17 +91,7 @@ def get_batch_size(train_dataloader):
...
-    elif trainer.accelerator == "ddp_cpu":
-        return trainer.num_devices * trainer.num_nodes
-    elif trainer.accelerator == "ddp":
-        return trainer.num_devices * trainer.num_nodes
-    else:
-        logging.warning(
-            f"The lightning trainer received accelerator: {trainer.accelerator}. We " "recommend to use 'ddp' instead."
-        )
-        return trainer.num_devices * trainer.num_nodes
+    return trainer.num_devices * trainer.num_nodes
```
Comment (titu1994) [inline nemo/collections/tts/helpers/helpers.py:94]: Can you check if trainer is none and return 1?

File: nemo/core/classes/modelPT.py [hunk 2]
```diff
@@ -483,21 +483,10 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
...
                 optim_config['sched']['t_accumulate_grad_batches'] = self._trainer.accumulate_grad_batches
                 optim_config['sched']['t_limit_train_batches'] = self._trainer.limit_train_batches
-                if self._trainer.accelerator is None:
-                    optim_config['sched']['t_num_workers'] = self._trainer.num_devices or 1
-                elif self._trainer.accelerator == "ddp_cpu":
-                    optim_config['sched']['t_num_workers'] = self._trainer.num_devices * self._trainer.num_nodes
-                elif self._trainer.accelerator == "ddp":
-                    optim_config['sched']['t_num_workers'] = self._trainer.num_devices * self._trainer.num_nodes
-                elif HAVE_NLPPLUGIN and isinstance(self._trainer.accelerator.training_type_plugin, NLPDDPPlugin):
+                optim_config['sched']['t_num_workers'] = self._trainer.num_devices * self._trainer.num_nodes
```
Comment (titu1994) [inline nemo/core/classes/modelPT.py:486]: Check if trainer is None and set to 1

File: nemo/core/classes/modelPT.py [hunk 3]
```diff
@@ -483,21 +483,10 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
...
+                if HAVE_NLPPLUGIN and isinstance(self._trainer.accelerator.training_type_plugin, NLPDDPPlugin):
                     app = AppState()
                     optim_config['sched']['t_num_workers'] = app.data_parallel_size
-                else:
-                    logging.warning(
-                        f"The lightning trainer received accelerator: {self._trainer.accelerator}. We "
-                        "recommend to use 'ddp' instead."
-                    )
-                    optim_config['sched']['t_num_workers'] = self._trainer.num_devices * self._trainer.num_nodes
```
Comment (titu1994) [inline nemo/core/classes/modelPT.py:500]: Hmm ok to remove the logging but I'd like a default else case

**Review by titu1994 — CHANGES_REQUESTED** [review]

File: nemo/collections/tts/helpers/helpers.py (hunk 1 above)
Comment (nithinraok) [reply nemo/collections/tts/helpers/helpers.py:94]: where would we use `get_num_workers` is trainer is None?

File: nemo/core/classes/modelPT.py (hunk 3 above)
Comment (redoctopus) [reply nemo/core/classes/modelPT.py:500]: Default case here would be to use the above calculation, do you have another behavior in mind?

File: nemo/collections/tts/helpers/helpers.py (hunk 1 above)
Comment (SeanNaren) [reply nemo/collections/tts/helpers/helpers.py:94]: Looking at the usage, both assume the trainer exists:
https://github.com/NVIDIA/NeMo/blob/b458394830719c1489261185ef728d0a2a5767a0/nemo/collections/tts/models/hifigan.py
https://github.com/NVIDIA/NeMo/blob/66ac0b3cc10d50a46635717bcfccb12426b8cefc/nemo/collections/tts/models/univnet.py

**Review by SeanNaren — APPROVED** [review]
I'm glad the abstractions in Lightning are cleaning up some of this code, was one of the motivations for cleaning up the API a bit :)

**Review by nithinraok — APPROVED** [review]

File: nemo/collections/tts/helpers/helpers.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/tts/helpers/helpers.py:94]: Alright if trainer is mandatory then this is fine as is.

File: nemo/core/classes/modelPT.py (hunk 3 above)
Comment (titu1994) [reply nemo/core/classes/modelPT.py:500]: Oh it moved up. My bad.

**Review by titu1994 — APPROVED** [review]
Lgtm, thanks

**Pull request merged**